pkg/app/piped/cloudprovider/terraform/module.go

```diff
@@ -0,0 +1,112 @@
+package terraform
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclparse"
+
+	"github.com/pipe-cd/pipecd/pkg/model"
+)
+
+// FileMapping is a schema for Terraform file.
+type FileMapping struct {
+	ModuleMappings []*ModuleMapping `hcl:"module,block"`
+	Remain         hcl.Body         `hcl:",remain"`
+}
+
+// ModuleMapping is a schema for "module" block in Terraform file.
+type ModuleMapping struct {
+	Name    string   `hcl:"name,label"`
+	Source  string   `hcl:"source"`
+	Version string   `hcl:"version"`
+	Remain  hcl.Body `hcl:",remain"`
+}
+
+// File represents a Terraform file.
+type File struct {
+	Modules []*Module
+}
+
+// Module represents a "module" block in Terraform file.
+type Module struct {
+	Name    string
+	Source  string
+	Version string
+}
+
+// LoadTerraformFiles loads terraform files from a given dir.
+func LoadTerraformFiles(dir string) ([]*File, error) {
```

Comment:
nit: I think we don't need a slice of pointers.
```suggestion
func LoadTerraformFiles(dir string) ([]File, error) {
```